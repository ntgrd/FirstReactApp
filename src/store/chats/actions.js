export const ADD_CHAT = "CHATS::ADD_CHAT";
export const DELETE_CHAT = "CHATS::DELETE_CHAT";

export const addChat = (chatName) => ({
    type: ADD_CHAT,
    payload: chatName,
});

export const deleteChat = (id) => ({
    type: DELETE_CHAT,
    payload: id,
});

