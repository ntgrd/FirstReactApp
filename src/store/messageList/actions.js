import {AUTHORS} from "../../utils/constants";

export const ADD_MESSAGE = "MESSAGES::ADD_MESSAGE";
export const DELETE_MESSAGE = "MESSAGES::DELETE_MESSAGE";

export const addMessage = (chatId, text, author) => ({
    type: ADD_MESSAGE,
    payload: {
        chatId,
        text,
        author,
    },
});

export const deleteMessage = (chatId, id) => ({
    type: DELETE_MESSAGE,
    payload: {
        chatId,
        id,
    },
});

let timer;
export const addMessageWithReply = (chatId, text, author) => (dispatch) => {
    dispatch(addMessage(chatId, text, author));
    if (author === AUTHORS.Natalia) {
        clearTimeout(timer);
        timer = setTimeout(() => {
            dispatch(addMessage(chatId, 'I am bot', AUTHORS.bot));
        }, 1500);
    }
};