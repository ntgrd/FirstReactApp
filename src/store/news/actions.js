import {PUBLIC_URL} from "../../utils/constants";

export const GET_ARTICLES_REQUEST = "ARTICLES::GET_REQUEST";
export const GET_ARTICLES_SUCCESS = "ARTICLES::GET_SUCCESS";
export const GET_ARTICLES_FAILURE = "ARTICLES::GET_FAILURE";

const gerArticlesRequest = () => ({
    type: GET_ARTICLES_REQUEST,
});

const gerArticlesSuccess = (articles) => ({
    type: GET_ARTICLES_SUCCESS,
    payload: articles
});

const gerArticlesFailure = (error) => ({
    type: GET_ARTICLES_FAILURE,
    payload: error
});
export const getArticles = () => async (dispatch) => {
    dispatch(gerArticlesRequest());
    try {
        const response = await fetch(PUBLIC_URL);

        if (!response.ok) {
            throw new Error(`Request failed with status ${response.status}`);
        }
        const result = await response.json();
        dispatch(gerArticlesSuccess(result));
    } catch (e) {
        console.log(e);
        dispatch(gerArticlesFailure(e.message));
    }
};
// export const getArticles = () => (dispatch) => {
//     dispatch(gerArticlesRequest());
//     fetch(PUBLIC_URL)
//         .then((response) => {
//             if (!response.ok) {
//                 throw new Error(`Request failed with status ${response.status}`);
//             }
//
//             return response.json();
//         })
//         .then((result) => {
//             dispatch(gerArticlesSuccess(result));
//         })
//         .catch((e) => {
//             console.log(e);
//             dispatch(gerArticlesFailure(e.message));
//         });
