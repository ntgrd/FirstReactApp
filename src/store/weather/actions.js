import {WEATHER_URL} from "../../utils/constants";

export const GET_WEATHER_REQUEST = "WEATHER::GET_REQUEST";
export const GET_WEATHER_SUCCESS = "WEATHER::GET_SUCCESS";
export const GET_WEATHER_FAILURE = "WEATHER::GET_FAILURE";

const gerWeatherRequest = () => ({
    type: GET_WEATHER_REQUEST,
});

const gerWeatherSuccess = (weather) => ({
    type: GET_WEATHER_SUCCESS,
    payload: weather
});

const getWeatherFailure = (error) => ({
    type: GET_WEATHER_FAILURE,
    payload: error
});
export const getWeather = () => async (dispatch) => {
    dispatch(gerWeatherRequest());
    try {
        const response = await fetch(WEATHER_URL);

        if (!response.ok) {
            throw new Error(`Request failed with status ${response.status}`);
        }
        const result = await response.json();
        dispatch(gerWeatherSuccess(result));
    } catch (e) {
        console.log(e);
        dispatch(getWeatherFailure(e.message));
    }
};