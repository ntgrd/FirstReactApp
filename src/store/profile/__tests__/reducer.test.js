import {profileReducer, initialState} from "../reducers";
import {changeName, toggleShowName} from "../actions";

describe('profileReducer', () => {
    it('returns correct state after SET_NAME action', () => {
        const userStore = profileReducer(initialState, changeName(`new name`));
        expect(userStore).toMatchSnapshot();
    });

    it('should return initial state for any unknown action', () => {
        const newState = profileReducer(undefined, {type: 'unknown'});
        expect(newState).toEqual(initialState);
    });

    it('should return state with new name for CHANGE_NAME action', () => {
        const newName = 'My New Name';
        const newState = profileReducer(undefined, changeName(newName));
        expect(newState.name).toBe(newName);
    });

    it('should toggle showName field for TOGGLE_SHOW_NAME action', () => {
        const newState = profileReducer(undefined, toggleShowName);
        expect(newState.showName).not.toBe(initialState.showName);
    });
});
