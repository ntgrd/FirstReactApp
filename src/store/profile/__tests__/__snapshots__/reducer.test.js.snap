// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profileReducer returns correct state after SET_NAME action 1`] = `
Object {
  "name": "new name",
  "showName": false,
}
`;
